### ISSUE TITLE
Python 2.4 compatibility broken (migrated from Trac #2071)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3618]
Original ticket http://projects.scipy.org/numpy/ticket/2071
Reported 2012-03-03 by atmention:rgommers, assigned to atmention:mwiebe.

{{{itertools.combinations}}} doesn't even exist in Python 2.5, so that should certainly be fixed.

```
======================================================================
ERROR: test_dtype.TestDtypeAttributeDeletion.test_dtype_non_writable_attributes_deletion
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/nose-1.1.2-py2.4.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/numpy/core/tests/test_dtype.py", line 412, in test_dtype_non_writable_attributes_deletion
    assert_raises(AttributeError, delattr, dt, s)
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/numpy/testing/utils.py", line 1053, in assert_raises
    return nose.tools.assert_raises(*args,**kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.4//lib/python2.4/unittest.py", line 320, in failUnlessRaises
    callableObj(*args, **kwargs)
TypeError: attribute 'subdtype' of 'numpy.dtype' objects is not writable

======================================================================
ERROR: test_maskna.test_array_maskna_view_dtype
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/nose-1.1.2-py2.4.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/numpy/core/tests/test_maskna.py", line 467, in test_array_maskna_view_dtype
    for x in itertools.combinations(tcs, 2):
AttributeError: 'module' object has no attribute 'combinations'

======================================================================
ERROR: test_nditer.test_iter_writable_attribute_deletion
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/nose-1.1.2-py2.4.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/numpy/core/tests/test_nditer.py", line 2564, in test_iter_writable_attribute_deletion
    assert_raises(AttributeError, delattr, it, s)
  File "/Library/Frameworks/Python.framework/Versions/2.4/lib/python2.4/site-packages/numpy/testing/utils.py", line 1053, in assert_raises
    return nose.tools.assert_raises(*args,**kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.4//lib/python2.4/unittest.py", line 320, in failUnlessRaises
    callableObj(*args, **kwargs)
TypeError: attribute 'value' of 'numpy.nditer' objects is not writable

----------------------------------------------------------------------
```

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-03-18

Should be fixed by PR 236

**Comment by thouis** [comment]
Comment in Trac by atmention:rgommers, 2012-03-19

Merged PR in commit:e7d3e33. 

And you turned a buildbot green! A Linux one, but still...

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-04-30
